src/TriggerBinding/SqlTableChangeMonitor.cs

```diff
@@ -0,0 +1,774 @@
+                            {
+                                await releaseLeasesCommand.ExecuteNonQueryAsync(token);
+                            }
+
+                            // Update the global state table if we have processed all changes with ChangeVersion <= newLastSyncVersion,
+                            // and clean up the worker table to remove all rows with ChangeVersion <= newLastSyncVersion.
+                            using (SqlCommand updateTablesPostInvocationCommand = this.BuildUpdateTablesPostInvocation(connection, transaction, newLastSyncVersion))
+                            {
+                                await updateTablesPostInvocationCommand.ExecuteNonQueryAsync(token);
+                            }
+
+                            transaction.Commit();
+                        }
+                        catch (Exception ex)
+                        {
+                            this._logger.LogError("Commit Exception Type: {0}", ex.GetType());
+                            this._logger.LogError("  Message: {0}", ex.Message);
+                            try
+                            {
+                                transaction.Rollback();
+                            }
+                            catch (Exception ex2)
+                            {
+                                this._logger.LogError("Rollback Exception Type: {0}", ex2.GetType());
+                                this._logger.LogError("  Message: {0}", ex2.Message);
+                            }
+                        }
+                    }
+                }
+
+            }
+            catch (Exception e)
+            {
+                // What should we do if releasing the leases fails? We could try to release them again or just wait,
+                // since eventually the lease time will expire. Then another thread will re-process the same changes
+                // though, so less than ideal. But for now that's the functionality.
+                this._logger.LogError($"Failed to release leases for user table {this._userTable.FullName} due to error: {e.Message}");
+            }
+            finally
+            {
+                // Want to do this before releasing the lock in case the renew leases thread wakes up. It will see that
+                // the state is checking for changes and not renew the (just released) leases.
+                await this.ClearRowsAsync(string.Empty, false);
+            }
+        }
+
+        /// <summary>
+        /// Calculates the new version number to attempt to update LastSyncVersion in global state table to. If all
+        /// version numbers in _rows are the same, use that version number. If they aren't, use the second largest
+        /// version number. For an explanation as to why this method was chosen, see 9c in Steps of Operation in this
+        /// design doc: https://microsoft-my.sharepoint.com/:w:/p/t-sotevo/EQdANWq9ZWpKm8e48TdzUwcBGZW07vJmLf8TL_rtEG8ixQ?e=owN2EX.
```

Comment:
Remove this - not public doc (and then add the relevant parts in comments below. The description shouldn't be in the summary since that's an implementation detail)